service/src/main/java/bio/terra/policy/service/pao/PaoService.java

```diff
@@ -256,21 +256,18 @@ public PolicyUpdateResult updatePao(
     // Now integrate the adds into the attribute set
     for (PolicyInput addPolicy : addAttributes.getInputs().values()) {
       PolicyInput existingPolicy = targetPao.getAttributes().lookupPolicy(addPolicy);
-      if (existingPolicy == null) {
-        // Nothing to combine; just take the new
-        attributesToUpdate.addInput(addPolicy);
-      } else {
-        PolicyInput addResult = PolicyMutator.combine(existingPolicy, addPolicy);
-        if (addResult != null) {
-          // We have a combined policy to add
+      PolicyInput addResult = PolicyMutator.combine(existingPolicy, addPolicy);
+      if (addResult != null) {
+        // We have a combined policy to add
+        if (existingPolicy != null) {
```

Comment:
Why do you need to remove here? The `addInput` will replace the existing attribute. "Add" is a bit of a misnomer. It is a put into a map. The only time we should have to remove is when there is no policy left.